libs/barista-components/autocomplete/src/autocomplete.ts

```diff
@@ -208,10 +210,14 @@ export class DtAutocomplete<T>
   private _projectedOptionsChangeSubscription: Subscription =
     Subscription.EMPTY;
 
+  /** Subject used for unsubscribing */
+  private _destroy$ = new Subject<void>();
+
   constructor(
     private _changeDetectorRef: ChangeDetectorRef,
     private _elementRef: ElementRef<HTMLElement>,
     private _viewContainerRef: ViewContainerRef,
+    private _ngZone: NgZone,
```

Comment:
@lukasholzer isn't this considered a breaking change? 
How should we handle this - we can't really go without the zone here - so there is no way around not having this functionality and no zone. One option would be to make the zone optional and not have the functionality when no zone is given. And make it mandatory in the next major version. - Should we add this to the breaking changes for 8? 
What do you think?

Reply:
Yea as it comes from the DI it would be ok only if you would extend from the autocomplete then it would break so we have to make it optional and mark it as deprecated to be non optional with version 9 I would say when we release 8 these days. 

I'm really not sure about how we should handle this (Can we mark the component class itself as internal?) so using it via DI is ok but if you want to extend it is on your own risk?

Because to be honest I don't want to support extending from the component.